core/domain/exp_services_test.py

```diff
@@ -2180,6 +2184,48 @@ def setUp(self):
 
         rights_manager.publish_exploration(self.ALBERT_ID, self.EXP_ID_2)
 
+        exploration_3 = self.save_new_valid_exploration(
+            self.EXP_ID_3, self.ALBERT_ID)
+
+        rights_manager.publish_exploration(self.ALBERT_ID, self.EXP_ID_3)
+
+        exp_services.delete_exploration(self.ALBERT_ID, self.EXP_ID_3)
+
+    def test_get_displayable_exploration_summary_dicts_matching_ids(self):
+        # A list of exp_id's are passed in:
+        # EXP_ID_1 -- private exploration
+        # EXP_ID_2 -- pubished exploration
+        # EXP_ID_3 -- deleted exploration
+        # Should only return [EXP_ID_2]
+
+        displayable_summaries = (
+            exp_services.get_displayable_exploration_summary_dicts_matching_ids(
+                [self.EXP_ID_1, self.EXP_ID_2, self.EXP_ID_3], self.ALBERT_ID))
+
+        self.assertEqual(len(displayable_summaries[0]), 11)
```

Comment:
I think there's a misinterpretation here. I don't think you should check the length of the dict; you already do that below, and in any case this check would be better phrased as checking the equality of the dict's keys with a given list.

What I was saying is: move line 2227 here.